### ISSUE TITLE
Tax Class error with ProductPost, but no tax codes exist

### ISSUE THREAD

**Issue by jlee2027** [bigcommerce/api#72]
re: store:
https://api.bigcommerce.com/stores/httiof4x/v2/tax_classes

We have converted 15 or so stores to the new V3 API. We find V3 to be a big step forward. Bravo! However, for **one store** on V3, we are unable to add products, getting a weird error.  Here is the JSON for one:

'{"name":"Bose 301 series II direct reflecting speakers pair 1982 brown/wood For repair","type":"physical","description":"Bose 301 Series II Direct Reflecting Speaker Pair FOR REPAIRVintage Bose Bookshelf Speakers8\\" subwoofer needs to be reconed, but worksHas direct reflecting tweeters on side which create a great, full soundWood grain exterior cabinetHas some chips, dings and scratchesSubwoofer sounds incredible, as does whole unitGreat project piece that still worksThey have been tested by our tech<br><br><font rwr=\'1\' size=\'4\' style=\'font-family:Arial\'><font rwr=\\"1\\" style=\\"font-family:Arial\\" size=\\"4\\"><font rwr=\\"1\\" style=\\"font-family:Arial\\" size=\\"4\\"><font rwr=\\"1\\" style=\\"font-family:Arial\\" size=\\"4\\"><b style=\\"color: rgb(81, 82, 81); font-family: &quot;PT Sans&quot;, Verdana, sans-serif; font-size: 16.002px; background-color: rgb(255, 255, 255);\\">Bose 301 Series II Direct Reflecting Speakers Pair FOR REPAIR<br><ul><li>Vintage Bose Bookshelf Speakers</li><li>8\\" subwoofer needs to be reconed, but works</li><li>Has direct reflecting tweeters on side which create a great, full sound</li><li>Wood grain exterior cabinet</li><li>Has some chips, dings and scratches</li><li>Subwoofer sounds incredible, as does whole unit</li><li>Great project piece that still works</li><li>They have been tested by our tech</li></ul></b></font></font></font></font>","sku":"201855918719","price":"115.00","inventory_tracking":"product","inventory_level":1,"availability":"available","weight":40,"upc":"","width":22,"height":22,"depth":12,"categories":["2612","2725","2726","2960"],"condition":"New","is_condition_shown":false,"is_visible":true,"custom_fields":[{"name":"Brand","value":"Bose"},{"name":"Model","value":"301 Series II"}],"variants":[]}'

**Response:**

ERROR:
array (
  'status' => 422,
  'title' => 'A tax class with id: 0 does not exist',
  'type' => 'https://developer.bigcommerce.com/api/status-codes',
  'errors' => 
  array (
    'tax_class_id' => 'A tax class with id: 0 does not exist',

The store has no tax codes.  https://api.bigcommerce.com/stores/httiof4x/v2/tax_classes returns a NULL

Please advise. Thanks

**Comment by bc-AlyssNoland** [comment]
@jlee2027 Are you still seeing this issue?

**Comment by jlee2027** [comment]
Yes. Any products we try to add or Quantity we change for just this one Store, returns the same error. More examples from a run I just now did to change QTY on existing products:

https://api.bigcommerce.com/stores/httiof4x/v3/catalog/products/59178
array (
  'status' => 422,
  'title' => 'A tax class with id: 0 does not exist',
  'type' => 'https://developer.bigcommerce.com/api#api-status-codes',
  'errors' => 
  array (
    'tax_class_id' => 'A tax class with id: 0 does not exist',
  ),
)https://api.bigcommerce.com/stores/httiof4x/v3/catalog/products/60867
array (
  'status' => 422,
  'title' => 'A tax class with id: 0 does not exist',
  'type' => 'https://developer.bigcommerce.com/api#api-status-codes',
  'errors' => 
  array (
    'tax_class_id' => 'A tax class with id: 0 does not exist',
  ),
)https://api.bigcommerce.com/stores/httiof4x/v3/catalog/products/60307
array (
  'status' => 422,
  'title' => 'A tax class with id: 0 does not exist',
  'type' => 'https://developer.bigcommerce.com/api#api-status-codes',
  'errors' => 
  array (
    'tax_class_id' => 'A tax class with id: 0 does not exist',
  ),
)https://api.bigcommerce.com/stores/httiof4x/v3/catalog/products/61242
array (
  'status' => 422,
  'title' => 'A tax class with id: 0 does not exist',
  'type' => 'https://developer.bigcommerce.com/api#api-status-codes',
  'errors' => 
  array (
    'tax_class_id' => 'A tax class with id: 0 does not exist',
  ),
)
